snapcraft/internal/yaml.py

```diff
@@ -109,15 +98,20 @@ def __init__(self, part):
 
 class Config:
 
+    @property
+    def part_names(self):
+        return self.parts_config._part_names
```

Comment:
Hmm... you're exporting something that's hidden in a subclass. That seems a little odd.